src/inspector_agent.cc

```diff
@@ -191,143 +193,117 @@ static int StartDebugSignalHandler() {
-                              receiver_(env->isolate(), receiver),
-                              callback_(env->isolate(), callback) {
-    session_.SetWeak(this, JsBindingsSessionDelegate::Release,
-                     v8::WeakCallbackType::kParameter);
-  }
-
-  ~JsBindingsSessionDelegate() override {
-    session_.Reset();
-    receiver_.Reset();
-    callback_.Reset();
-  }
+  class JSBindingsSessionDelegate : public InspectorSessionDelegate {
+   public:
+    JSBindingsSessionDelegate(Environment* env,
+                              JSBindingsConnection* connection)
+                              : env_(env),
+                                connection_(connection) {
+    }
 
-  bool WaitForFrontendMessageWhilePaused() override {
-    return false;
-  }
+    bool WaitForFrontendMessageWhilePaused() override {
+      return false;
+    }
 
-  void SendMessageToFrontend(const v8_inspector::StringView& message) override {
-    Isolate* isolate = env_->isolate();
-    v8::HandleScope handle_scope(isolate);
-    Context::Scope context_scope(env_->context());
-    MaybeLocal<String> v8string =
-        String::NewFromTwoByte(isolate, message.characters16(),
-                               NewStringType::kNormal, message.length());
-    Local<Value> argument = v8string.ToLocalChecked().As<Value>();
-    Local<Function> callback = callback_.Get(isolate);
-    Local<Object> receiver = receiver_.Get(isolate);
-    callback->Call(env_->context(), receiver, 1, &argument)
-        .FromMaybe(Local<Value>());
-  }
+    void SendMessageToFrontend(const v8_inspector::StringView& message)
+        override {
+      Isolate* isolate = env_->isolate();
+      v8::HandleScope handle_scope(isolate);
+      Context::Scope context_scope(env_->context());
+      MaybeLocal<String> v8string =
+          String::NewFromTwoByte(isolate, message.characters16(),
+                                 NewStringType::kNormal, message.length());
+      Local<Value> argument = v8string.ToLocalChecked().As<Value>();
+      connection_->OnMessage(argument);
+    }
 
-  void Disconnect() {
-    Agent* agent = env_->inspector_agent();
-    if (agent->delegate() == this)
-      agent->Disconnect();
-  }
+    void Disconnect() {
+      Agent* agent = env_->inspector_agent();
+      if (agent->delegate() == this)
+        agent->Disconnect();
+    }
 
- private:
-  static void Release(
-      const v8::WeakCallbackInfo<JsBindingsSessionDelegate>& info) {
-    info.SetSecondPassCallback(ReleaseSecondPass);
-    info.GetParameter()->session_.Reset();
+   private:
+    Environment* env_;
+    JSBindingsConnection* connection_;
+  };
+
+  JSBindingsConnection(Environment* env,
+                       Local<Object> wrap,
+                       Local<Function> callback)
+                       : AsyncWrap(env, wrap, PROVIDER_INSPECTORJSBINDING),
+                         delegate_(env, this),
+                         callback_(env->isolate(), callback) {
```

Comment:
style nit: mind changing the spacing of the initialization list to match that of `NodeInspectorClient()`?

Reply:
I thought this was the predominant style for initializers, and I also prefer this style a bit more.

There are other places in the file where the initializer list is in this style. I'd be happy to make this consistent in a separate PR if you insist, but I think this change should get in now separated from stylistic changes..